src/SHA.jl

```diff
@@ -51,7 +51,7 @@ for (f, ctx) in [(:sha1, :SHA1_CTX),
         # open("test.txt") do f
         #     sha256(f)
         # done
-        $f(io::IO) = $f(readbytes(io))
+        $f(io::IO) = @compat $f(read(io))
```

Comment:
Wouldn't it be better to have `@compat $f(read(io, Array{UInt8, 1}, 1))`? It would explicitly define that we need an `UInt8` array, with one dimension, IMO.